### PR TITLE
It stores the template string so components can be extended over the …

### PR DESCRIPTION
**Pull request by dmarcos**
…base template. It also adds an extendable flag so a component can be declared as non extendable and the template string is not stored

### CODE REVIEW

File: gaia-component.js [hunk 1]
```diff
@@ -20,13 +20,26 @@ var noop  = function() {};
  */
 exports.register = function(name, props) {
   var baseProto = getBaseProto(props.extends);
+  var template = props.template || baseProto.templateString;
+
+  // Components are extendable by default but can be declared
+  // as non extendable as an optimization to avoid
+  // storing the template strings
+  var extendable = props.extendable =
+    props.hasOwnProperty('extendable')? props.extendable : true;
```
Comment (wilsonpage) [inline gaia-component.js]: NIT 

``` js
var extendable = props.extendable = props.hasOwnProperty('extendable') ?
  props.extendable : true;
```

**Comment by dmarcos** [comment]
@wilsonpage Tests are in

File: test/test.js [hunk 2]
```diff
@@ -197,6 +197,38 @@ suite('gaia-component', function() {
...
+    test('it does not store the template string if the component is declared as non extensible', function() {
+      var El = component.register('test-extends-non-extensible', {
+        extends: HTMLInputElement,
+        extensible: false,
+        template: `<div class="inner"></div>`
+      });
+      assert.isUndefined(El.prototype.templateString);
+    });
+
+    test('it does not store the template string if the component is declared as non extensible', function() {
```
Comment (wilsonpage) [inline test/test.js]: Typo: Test name needs updating.

**Comment by wilsonpage** [comment]
Just a couple of linting errors:

``` bash
test/test.js: line 209, col 109, Line is too long.
test/test.js: line 218, col 109, Line is too long.
test/test.js: line 226, col 93, Line is too long.
test/test.js: line 229, col 110, Line is too long.
```

And need to update a test name. 

Thanks man, this is really valuable :)

**Comment by dmarcos** [comment]
@wilsonpage done

**Pull request merged**